Troubleshoot/TroubleshootError.ps1

```diff
@@ -231,7 +262,104 @@ else {
     }
 }
 
-if ($LogAnalyticsWorkspaceResourceID -eq $null) {
+
+Write-Host("Currently checking if the cluster is onboarded to custom metrics for Azure monitor for containers...");
+
+#Pre requisite - need cluster spn object Id
+$clusterDetails = Get-AzureRmAks -Id $AKSClusterResourceId -ErrorVariable clusterFetchError -ErrorAction SilentlyContinue;
+if ($clusterFetchError) {
+    Write-Host("Error in fetching Cluster details for " + $AKSClusterName) -ForegroundColor Red;
+    Write-Host("Please contact us by emailing askcoin@microsoft.com for help") -ForegroundColor Red;
+    Write-Host("");
+}
+else {
+    Write-Host($clusterDetails | Format-List | Out-String);
+    $clusterSPNClientID = $clusterDetails.ServicePrincipalProfile.ClientId;
+    $clusterLocation = $clusterDetails.Location;
+
+    if ($MdmCustomMetricAvailabilityLocations -contains $clusterLocation) {
+        Write-Host('Cluster is in a location where Custom metrics are available') -ForegroundColor Green;
+        if ($null -eq $clusterSPNClientID ) {
+            Write-Host("There is no service principal associated with this cluster.") -ForegroundColor Red;
+            Write-Host("Please contact us by emailing askcoin@microsoft.com for help") -ForegroundColor Red;
+            Write-Host("");
+        }
+        else {
+            # Convert the client ID to the Object ID
+            $clusterSPN = Get-AzureRmADServicePrincipal -ServicePrincipalName $clusterSPNClientID;
+            $clusterSPNObjectID = $clusterSPN.Id;
+            if ($null -eq $clusterSPNObjectID) {
+                Write-Host("Couldn't convert Client ID to Object ID.") -ForegroundColor Red;
+                Write-Host("Please contact us by emailing askcoin@microsoft.com for help") -ForegroundColor Red;
+                Write-Host("");
+            }
+
+            $MonitoringMetricsPublisherCandidates = Get-AzureRmRoleAssignment -RoleDefinitionName $MonitoringMetricsRoleDefinitionName -Scope $AKSClusterResourceId -ErrorVariable notPresent -ErrorAction SilentlyContinue
+
+            if ($notPresent) {
+                Write-Host("Error in fetching monitoring metrics publisher candidates for " + $AKSClusterName) -ForegroundColor Red;
+                Write-Host("Please contact us by emailing askcoin@microsoft.com for help") -ForegroundColor Red;
```

Comment:
This might be permission issue we should throw an error  and exit.

Reply:
updated in PR into ci_feature : abandoning this PR:
https://github.com/Microsoft/OMS-docker/pull/243

I donot want to exit and stop the script here since the other workspace troubleshooting part hasn't run yet.